src/signature_mahalanobis_knn/sig_mahal_knn.py

```diff
@@ -24,6 +24,7 @@ def __init__(
         """
         self.signature_transform = None
         self.n_jobs = n_jobs
+        self.mahal_distance = None
```

Comment:
Do we need the SigMahaKNN object to hold the distance object? What are the situations that we need to use it directly outside of the class?

Reply:
With Jonathan, we added this in for debugging mainly, it was very useful to have it and you to use the distance object directly and to compute a distance rather than just the KNN to the corpus.

Reply:
Another reason is that for pynndescent to work, we can only use abstract methods (we have to use the jitted `Mahalanobis.calc_distance` rather than `Mahalanobis.distance`) as it doesn't like having `self` as an argument. There may be other ways around this.

But in short, we need access to the attributes in the distance object to make pynndescent work.